catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -161,6 +160,29 @@ private static double metersToDegrees(double distance) {
     }
 
     @Override
+    public SolrQuery propertyIsEqualTo(String functionName, List<Object> arguments,
+            Object literal) {
+        String not;
+        SolrQuery query;
+        switch (functionName) {
+        case "divisibleBy":
+
+            //the return type is boolean so cast the literal to boolean and in effect this is just a NOT so we will update the query as such9
+            not = (Boolean) literal ? "" : "!";
+            query = propertyIsDivisibleBy((String) arguments.get(0), (Long) arguments.get(1));
+            return query.setQuery(not + query.getQuery());
+        case "proximity":
+            not = (Boolean) literal ? "" : "!";
```

Comment:
hmm its not really common logic that literal could be anything its just for these functions the return type they dictate is a boolean and so I can do a not on their result.